docs/content/direct/common-setup-hub-chart.md

```diff
@@ -0,0 +1,80 @@
+# Common Setup - hub chart
+
+For more details please see [here](./hub-chart.md).
+
+This documents explains how to use KubeStellar Hub chart to deploy a new instance of KubeStellar
+with a choice of user-defined KubeFlex Control Planes (CPs).
+
+The information provided is specific for the following release:
+
+```shell
+export KUBESTELLAR_VERSION=0.23.0-alpha.2
+```
+
+## Create a Kind cluster
+
+The setup of KubStellar via the Hub chart requires the existance of at least one cluster
+to be used for the deployment of the chart.
+
+For convenience, a new local Kind cluster that satisfies the requirements for KubeStellar setup
+and that can be used to exercises the [examples](./examples.md) can be created with the following command:
+
+```shell
+sudo su <<EOF
+bash <(curl -s https://raw.githubusercontent.com/kubestellar/kubestellar/v${KUBESTELLAR_VERSION}/scripts/create-kind-cluster-with-SSL-passthrough.sh) --name kubeflex --port 9443
+EOF
+```
+
+## Install KubeStellar Hub chart
+
+A KubeStellar Hub installation compatible with the common setup suitable for Common Setup described in the [examples](https://docs.kubestellar.io/release-0.22.0/direct/examples/) could be achieved with the following command:
+
+```shell
+helm upgrade --install ks-hub oci://ghcr.io/kubestellar/kubestellar/hub-chart --version $KUBESTELLAR_VERSION \
+  --set-json='ITSes=[{"name":"its1"}]' \
+  --set-json='WDSes=[{"name":"wds1"}]'
+```
+
+Remember to include the option `--set "kubeflex-operator.isOpenShift=true"`, when deploying in an OpenShift cluster.
+
+## Obtain the kubeconfigs to access the KubeFlex Control Planes
+
+After the KubeStellar Hub has fully deployed,
+the following code can be used to obtain the kubeconfig of all the KubeFlex Control Planes in the hub cluster:
+
+```shell
+echo "Getting the kubeconfig of all Control Planes..."
+for cpname in `kubectl get controlplane -o name`; do

```

Comment:
This makes a big new assumption: that the user does not have any irrelevant control planes, or that they are sufficiently few/inconsequential that this overspray is OK.

Reply:
And?  What's the problem in that ?  It is clearly said it takes ALL the CPs defined here.  If users want to take this as an example but retrieve a subset of CPs they can do that themselves.

Reply:
"defined here"? What do you mean by "here"? The reader's attention is focused on the control planes involved in the Helm chart. But surprise! Now we get a command that deals with all the control planes in KubeFlex, regardless of where they came from or what they are for.

There is a non-small amount of bash code in the loop body here; if the user wants to apply it in any other way than being in this loop then there is a lot to copy and adapt.

Reply:
@MikeSpreitzer I cannot cover all possible scenarios, I think.
This is intended to teach a user how to get a CP kubeconfig and provide an easy copy&paste way to do them all.
In you special cases you can choose what to do.
@clubanderson is already doing this in his cluster/setup... we need to get to him to see what he does.

I think the most important decision is to swithc to use separated kubeconfigs or continue merging into a single one and use contexts... this strongly impacts the rewording of the examples

Note that merging is not a good idea in general, for example all its configs have the same vcluster server name... there will be a conflict

Reply:
Why this step of obtaining all the kubeconfigs for all CPs and saving them in separate files in the current folder is required and how does it relate to the single chart ?

Reply:
This is the future version of the examples.md file content. So the setup has to connect with the usage in the example scenarios.

Reply:
@pdettori: This is the future version of the examples.md file content. So the setup has to connect with the usage in the example scenarios.

@francostellari: what about using `kflex` to do the kubeconfig updates?

Reply:
This is in the context of the example setup, where the set of control planes is fixed. This line could simply be:

```
for cpname in wds1 its1; do
```

Reply:
ok